@here/harp-mapview/lib/Utils.ts

```diff
@@ -146,22 +147,35 @@ export namespace MapViewUtils {
         pitchDeg: number,
         mapView: MapView
     ): geoUtils.GeoCoordinates {
-        // Get the world distance between the target and the camera.
         const pitchRad = THREE.Math.degToRad(pitchDeg);
-        const cameraHeight = distance * Math.cos(pitchRad);
-        const projectedDistanceOnTheGround = cameraHeight * Math.tan(pitchRad);
-
-        // Get the camera coordinates.
         const yawRad = THREE.Math.degToRad(yawDeg);
-        const worldTargetCoordinates = mapView.projection.projectPoint(targetCoordinates);
-        const cameraWorldCoordinates = {
-            x: worldTargetCoordinates.x + Math.sin(yawRad) * projectedDistanceOnTheGround,
-            y: worldTargetCoordinates.y - Math.cos(yawRad) * projectedDistanceOnTheGround,
-            z: 0
-        };
+        mapView.projection.projectPoint(targetCoordinates, vector);
+        const groundDistance = distance * Math.sin(pitchRad);
+        if (mapView.projection.type === geoUtils.ProjectionType.Planar) {
+            vector.set(
+                vector.x + Math.sin(yawRad) * groundDistance,
+                vector.y - Math.cos(yawRad) * groundDistance,
+                0
+            );
+        } else if (mapView.projection.type === geoUtils.ProjectionType.Spherical) {
+            // In globe yaw and pitch are understood to be in tangent space. The approach below is
+            // to find the Z and Y tangent space axes, then rotate Y around Z by the given yaw, and
+            // set its new length (groundDistance). Finally the up vector's length is set to the
+            // camera height and added to the transformed Y above.
+            tangentSpace.z.copy(vector).normalize();
+            const obj = new THREE.Object3D();
```

Comment:
Why is an `Object3D` needed here? You can use a regular `Matrix4` to do the `lookAt` and the world direction can be easily extracted from that matrix like this https://github.com/mrdoob/three.js/blob/master/src/core/Object3D.js#L563